UserContext/AnonymousRequestMatcher.php

```diff
@@ -1,47 +0,0 @@
-<?php
-
-/*
- * This file is part of the FOSHttpCacheBundle package.
- *
- * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
- *
- * For the full copyright and license information, please view the LICENSE
- * file that was distributed with this source code.
- */
-
-namespace FOS\HttpCacheBundle\UserContext;
-
-use Symfony\Component\HttpFoundation\Request;
-use Symfony\Component\HttpFoundation\RequestMatcherInterface;
-
-/**
- * Matches anonymous requests using a list of identification headers.
- */
-class AnonymousRequestMatcher implements RequestMatcherInterface
```

Comment:
for BC, we need to keep this class around. i think the best is to make it extend the one from the library but keep the constructor signature, and forward the headers as the options array. and trigger a deprecation warning.

imo its ok to make the service definition use the library matcher directly. my thing is for people who use this class to define their own service.